pandas/core/nanops.py

```diff
@@ -479,6 +479,9 @@ def nanvar(values, axis=None, skipna=True, ddof=1):
 
 @disallow('M8', 'm8')
 def nansem(values, axis=None, skipna=True, ddof=1):
+    # This checks if non-numeric-like data is passed with numeric_only=False
+    # and raises a TypeError otherwise
+    var = nanvar(values, axis, skipna, ddof=ddof)
```

Comment:
If `var` isn't used, don't assign the result of `nanvar` to a variable.  Just call it without.